File: Mntone.SplatoonClient/Mntone.SplatoonClient/Entities/Intention.cs
using System;
using System.Runtime.Serialization;

namespace Mntone.SplatoonClient.Entities
{
	/// <summary>
	/// Intention information
	/// </summary>
	[DataContract]
	public sealed class Intention
	{
		internal Intention(string id, Uri imageUri)
		{
			this.ID = id;
			this.ImageUri = imageUri;
		}

		/// <summary>
		/// Intention ID
		/// </summary>
		[DataMember(Name = "id")]
		public string ID { get; private set; }

		/// <summary>
		/// Image uri
		/// </summary>
		[DataMember(Name = "image")]
		public Uri ImageUri { get; private set; }
	}
}

File: net.mntone.splatoonclient/net.mntone.splatoonclient.entities/src/main/java/net/mntone/splatoonclient/entities/Intention.java
package net.mntone.splatoonclient.entities;

import org.json.JSONObject;

import java.net.MalformedURLException;
import java.net.URL;

public final class Intention
{
	private final String _id;
	private final URL _uri;

	Intention(final String id, final URL uri)
	{
		this._id = id;
		this._uri = uri;
	}

	Intention(final JSONObject json) throws MalformedURLException
	{
		this._id = json.getString("id");
		this._uri = new URL(json.getString("image"));
	}

	public String getId()
	{
		return this._id;
	}

	public URL getUri()
	{
		return this._uri;
	}
}